src/logs.ts

```diff
@@ -0,0 +1,225 @@
+      consoleLog: Logs.console.ok,
+      logMessage: log,
+      metadata,
+      type: "ok",
+    });
+  }
+
+  public info(log: string, metadata?: Metadata): LogReturn | null {
+    metadata = this._addDiagnosticInformation(metadata);
+    return this._log({
+      level: LOG_LEVEL.INFO,
+      consoleLog: Logs.console.info,
+      logMessage: log,
+      metadata,
+      type: "info",
+    });
+  }
+
+  public error(log: string, metadata?: Metadata): LogReturn | null {
+    metadata = this._addDiagnosticInformation(metadata);
+    return this._log({
+      level: LOG_LEVEL.ERROR,
+      consoleLog: Logs.console.error,
+      logMessage: log,
+      metadata,
+      type: "error",
+    });
+  }
+
+  public debug(log: string, metadata?: Metadata): LogReturn | null {
+    metadata = this._addDiagnosticInformation(metadata);
+    return this._log({
+      level: LOG_LEVEL.DEBUG,
+      consoleLog: Logs.console.debug,
+      logMessage: log,
+      metadata,
+      type: "debug",
+    });
+  }
+
+  public fatal(log: string, metadata?: Metadata): LogReturn | null {
+    if (!metadata) {
+      metadata = Logs.convertErrorsIntoObjects(new Error(log)) as Metadata;
+      const stack = metadata.stack as string[];
+      stack.splice(1, 1);
+      metadata.stack = stack;
+    }
+
+    if (metadata instanceof Error) {
+      metadata = Logs.convertErrorsIntoObjects(metadata) as Metadata;
+      const stack = metadata.stack as string[];
+      stack.splice(1, 1);
+      metadata.stack = stack;
+    }
+
+    metadata = this._addDiagnosticInformation(metadata);
+    return this._log({
+      level: LOG_LEVEL.FATAL,
+      consoleLog: Logs.console.fatal,
+      logMessage: log,
+      metadata,
+      type: "fatal",
+    });
+  }
+
+  public verbose(log: string, metadata?: Metadata): LogReturn | null {
+    metadata = this._addDiagnosticInformation(metadata);
+    return this._log({
+      level: LOG_LEVEL.VERBOSE,
+      consoleLog: Logs.console.verbose,
+      logMessage: log,
+      metadata,
+      type: "verbose",
+    });
+  }
+
+  constructor(logLevel: LogLevel) {
+    this._maxLevel = this._getNumericLevel(logLevel);
+    Logs.console = new PrettyLogs();
+  }
+
+  static _commentMetaData(metadata: Metadata, level: LogLevel) {
+    Logs.console.debug("the main place that metadata is being serialized as an html comment");
+    const prettySerialized = JSON.stringify(metadata, null, 2);
+    // first check if metadata is an error, then post it as a json comment
+    // otherwise post it as an html comment
+    if (level === LOG_LEVEL.FATAL) {
+      return ["```json", prettySerialized, "```"].join("\n");
+    } else {
+      return ["<!--", prettySerialized, "-->"].join("\n");
+    }
+  }
+
+  private _diffColorCommentMessage(type: string, message: string) {
+    const diffPrefix = {
+      fatal: "-", // - text in red
+      ok: "+", // + text in green
+      error: "!", // ! text in orange
+      // info: "#", // # text in gray
```

Comment:
Why all the comments?

Reply:
These are comments from the original I thought best to leave them but I will do a thorough clean up of those that are not needed